kore/src/Kore/Validate/PatternVerifier.hs

```diff
@@ -230,15 +232,39 @@ verifyOperands operandSort = \operator -> do
     traverse verifyChildWithSort operator
 {-# INLINE verifyOperands #-}
 
+internalizeAnd ::
+    And Sort Verified.Pattern ->
+    BinaryAnd Sort Verified.Pattern
+internalizeAnd And{andSort, andChildren = [andFirst, andSecond]} =
+    BinaryAnd{andSort, andFirst, andSecond}
+internalizeAnd And{andSort, andChildren} =
+    let first = head andChildren
+     in let second = internalizeAnd And{andSort, andChildren = tail andChildren}
+         in BinaryAnd{andSort, andFirst = first, andSecond = synthesize (Internal.AndF second)}
+
+internalizeOr ::
+    Or Sort Verified.Pattern ->
+    BinaryOr Sort Verified.Pattern
+internalizeOr Or{orSort, orChildren = [orFirst, orSecond]} =
+    BinaryOr{orSort, orFirst, orSecond}
+internalizeOr Or{orSort, orChildren} =
+    let first = head orChildren
+     in let second = internalizeOr Or{orSort, orChildren = tail orChildren}
+         in BinaryOr{orSort, orFirst = first, orSecond = synthesize (Internal.OrF second)}
+
 verifyAnd ::
     And Sort (PatternVerifier Verified.Pattern) ->
-    PatternVerifier (And Sort Verified.Pattern)
-verifyAnd = verifyOperands andSort
+    PatternVerifier (BinaryAnd Sort Verified.Pattern)
+verifyAnd operands = do
+    verifiedAnd <- verifyOperands And.andSort operands
+    return (internalizeAnd verifiedAnd)
```

Comment:
Actually, alternatively to the suggestion above, we could verify that the And has at lease two children here. I think I prefer this option. Please go with the one you find most  appropriate. The same will need to be done for Or.
```suggestion
    koreFailWhen (length operands < 2) "Cannot internalize And with less than two children"
    verifiedAnd <- verifyOperands And.andSort operands
    return (internalizeAnd verifiedAnd)
```